aspnetcore/data/ef-rp/intro/samples/cu50/Pages/Instructors/Create.cshtml.cs

```diff
@@ -32,27 +39,44 @@ public IActionResult OnGet()
         public async Task<IActionResult> OnPostAsync(string[] selectedCourses)
         {
             var newInstructor = new Instructor();
-            if (selectedCourses != null)
+            newInstructor.Courses = new List<Course>();
+
+            // Add selected Courses courses to the new instructor.
+            foreach (var course in selectedCourses)
             {
-                newInstructor.Courses = new List<Course>();
-                foreach (var course in selectedCourses)
+                var foundCourse = await _context.Courses.FirstOrDefaultAsync(
```

Comment:
I would use `Find` here instead of `First`. This way the course won't be loaded again if it's already being tracked. If the courses collection isn't large, you could even preemptively do a `_context.Courses.Load()` prior to the loop.